### PR TITLE
Torch baselines for `batchnorm` and `instance_norm`

### PR DESCRIPTION
**Pull request by Priya2698**
Issue #1668

### CODE REVIEW

**Comment by Priya2698** [comment]
> can you add comments about which dim is normalized and which scheduler is being used?

Is there a DEBUG_DUMP option to print the scheduler used or I'll need to debug statements to infer?

**Comment by liqiangxl** [comment]
> > can you add comments about which dim is normalized and which scheduler is being used?
> 
> Is there a DEBUG_DUMP option to print the scheduler used or I'll need to debug statements to infer?

you can use NVFUSER_DUMP=scheduler_params
For normalization we have: inner persistent (2D or 3D), outer persistent, innerOuter persistent
IIRC, instance norm is outer persistent and batch norm is inner persistent 3D.

**Comment by Priya2698** [comment]
> > > can you add comments about which dim is normalized and which scheduler is being used?
> > 
> > 
> > Is there a DEBUG_DUMP option to print the scheduler used or I'll need to debug statements to infer?
> 
> you can use NVFUSER_DUMP=scheduler_params For normalization we have: inner persistent (2D or 3D), outer persistent, innerOuter persistent IIRC, instance norm is outer persistent and batch norm is inner persistent 3D.

Thanks.
While looking into the schedulers used for layernorm, it used reduction kernel for larger sizes.

**Comment by liqiangxl** [comment]
> > > > can you add comments about which dim is normalized and which scheduler is being used?
> > > 
> > > 
> > > Is there a DEBUG_DUMP option to print the scheduler used or I'll need to debug statements to infer?
> > 
> > 
> > you can use NVFUSER_DUMP=scheduler_params For normalization we have: inner persistent (2D or 3D), outer persistent, innerOuter persistent IIRC, instance norm is outer persistent and batch norm is inner persistent 3D.
> 
> Thanks. While looking into the schedulers used for layernorm, it used reduction kernel for larger sizes. The inner outer kernel was used in the smaller ranges.

If segmented, it uses a reduction kernel followed by a pointwise.

**Comment by Priya2698** [comment]
@liqiangxl 
For a subset of the inputs, I saw the following schedulers:
1. Reduction with tag Outer persistent
2. Reduction with no tags
3. Reduction + Pointwise

Since there are large number of inputs, I haven't verified it for all input sizes.
What is the motivation behind listing the scheduler? It is not feasible to do it atleast manually.

**Comment by liqiangxl** [comment]
> @liqiangxl For a subset of the inputs, I saw the following schedulers:
> 
> 1. Reduction with tag Outer persistent
> 2. Reduction with no tags
> 3. Reduction + Pointwise
> 
> Since there are large number of inputs, I haven't verified it for all input sizes. What is the motivation behind listing the scheduler? It is not feasible to do it atleast manually.

It seems that we need to clean the scheduler tags. The motivation is to know which scheduler is being used so we can quickly locate the issue if there is a regression and we may also add a feature to select a subset of benchmarks based on the scheduler. For example, if I modified the inner persistent scheduler, I probably only want to run benchmarks using the inner persistent scheduler.
Since this info is also missing in other benchmark cases, we can discuss it later and add in another PR. @naoyam what do you think about ` add a feature to select a subset of benchmarks based on the scheduler`?

**Review by liqiangxl — APPROVED** [review]
LGTM. discuss adding comments and tags about which scheduler is being used.

**Comment by Priya2698** [comment]
> > @liqiangxl For a subset of the inputs, I saw the following schedulers:
> > 
> > 1. Reduction with tag Outer persistent
> > 2. Reduction with no tags
> > 3. Reduction + Pointwise
> > 
> > Since there are large number of inputs, I haven't verified it for all input sizes. What is the motivation behind listing the scheduler? It is not feasible to do it atleast manually.
> 
> It seems that we need to clean the scheduler tags. The motivation is to know which scheduler is being used so we can quickly locate the issue if there is a regression and we may also add a feature to select a subset of benchmarks based on the scheduler. For example, if I modified the inner persistent scheduler, I probably only want to run benchmarks using the inner persistent scheduler. Since this info is also missing in other benchmark cases, we can discuss it later and add in another PR. @naoyam what do you think about ` add a feature to select a subset of benchmarks based on the scheduler`?

How would such filtering work? How can we know upfront which inputs will use a given scheduler?
